Source/TrackableEntities.EF.5/DbContextExtensions.cs

```diff
@@ -199,16 +199,31 @@ public static void ApplyChanges(this InterceptorPool pool, IEnumerable<ITrackabl
                     context.ApplyChangesOnProperties(item, visitationHelper.Clone(), TrackingState.Deleted, interceptors);
                 }
 
+                var isComplex = context.IsComplexType(item.GetType());
+                bool isParentEligible(TrackingState parentState) => parentState == TrackingState.Modified || parentState == TrackingState.Unchanged;
+                var isModifiable = 
+                          (isComplex 
+                            ? isParentEligible(parent.TrackingState) 
+                            : (item.TrackingState == TrackingState.Modified && (state == null || state == TrackingState.Modified)))                          
+                          && item.ModifiedProperties?.Count > 0;
+
                 // Set modified properties
-                if (item.TrackingState == TrackingState.Modified
-                    && (state == null || state == TrackingState.Modified)
-                    && item.ModifiedProperties != null
-                    && item.ModifiedProperties.Count > 0)
+                if (isModifiable)
                 {
                     // Mark modified properties
                     SetEntityState(context, item, parent, propertyName, EntityState.Unchanged, interceptors);
                     foreach (var property in item.ModifiedProperties)
-                        context.Entry(item).Property(property).IsModified = true;
+                    {
+                        if(isComplex)
+                        {
+                            var parentEntry = context.Entry(parent);
+                            if(parentEntry.State == EntityState.Detached)
+                              parentEntry.State = parent.TrackingState.ToEntityState();
+                            parentEntry.ComplexProperty(propertyName).Property(property).IsModified = true;                           
+                        }
+                        else
+                            context.Entry(item).Property(property).IsModified = true;
+                    }
```

Comment:
Actually, `DbComplexPropertyEntry` inherits from `DbPropertyEntry`, so you could clean this up be adding a getPropertyEntry method that returns `DbPropertyEntry` and setting `IsModified` to true.

```csharp
DbPropertyEntry GetPropertyEntry(...)
// if complex type return DbComplexPropertyEntry
```